### PR TITLE
✨ Add Support for Ironic Basic auth in deployment templates

### PR DESCRIPTION
**Pull request by maelk**
**What this PR does / why we need it**:
This PR adds support for basic authentication towards Ironic and Ironic inspector.

Based on #133

### CODE REVIEW

**Comment by maelk** [comment]
The tests here will not pass until the metal3-dev-env modifications are done

File: config/bmo/kustomization.yaml [hunk 1]
```diff
@@ -13,11 +14,13 @@ configMapGenerator:
   - DEPLOY_RAMDISK_URL=${DEPLOY_RAMDISK_URL}
   - IRONIC_ENDPOINT=${IRONIC_URL}
   - IRONIC_INSPECTOR_ENDPOINT=${IRONIC_INSPECTOR_URL}
+  - IRONIC_CACERT_FILE=/opt/metal3/certs/ca/tls.crt
```
Comment (fmuyassarov) [inline config/bmo/kustomization.yaml:17]: Can't we parameterize it as any other vars here?

File: docs/getting-started.md [hunk 2]
```diff
@@ -26,30 +26,135 @@ additional information on the `move` process.
...
+
+```sh
+  export IRONIC_INSPECTOR_PASSWORD="<password>"
+```
+
+##### IRONIC_INSPECTOR_NO_BASIC_AUTH
+
+Disables basic authentication for Ironic inspector API. Any value provided in
+this variable disables authentication. To enable authentication, leave this
+variable unset. If unset, then `IRONIC_USERNAME` and `IRONIC_PASSWORD` must be
```
Comment (fmuyassarov) [inline docs/getting-started.md:152]: Should they instead be `IRONIC_INSPECTOR_PASSWORD` and `IRONIC_INSPECTOR_USERNAME` ?

**Review by fmuyassarov — COMMENTED** [review]
Looks good.

File: config/bmo/kustomization.yaml (hunk 1 above)
Comment (maelk) [reply config/bmo/kustomization.yaml:17]: No this is within the pod. What matters is the secret we mount there. And we create it based on the Base64 of the certificate

File: docs/getting-started.md (hunk 2 above)
Comment (maelk) [reply docs/getting-started.md:152]: good catch

**Pull request merged**